Bug in celery.contrib.testing.worker.start_worker

When `_start_worker_thread` fails, the code the `finally` fails as well because `worker` is not defined.

https://github.com/celery/celery/blob/f4628639ca39be4d0aea39ec1c2f066117529316/celery/contrib/testing/worker.py#L75-L91

Not a big deal but confusing when debugging.

Traceback:

```
app = <Celery celery.tests at 0x7f57bdd6ee50>, concurrency = 1, pool = 'prefork'
loglevel = 'debug', logfile = None, perform_ping_check = True
ping_task_timeout = 10.0, shutdown_timeout = 10.0, kwargs = {}
    @contextmanager
    def start_worker(
        app,  # type: Celery
        concurrency=1,  # type: int
        pool='solo',  # type: str
        loglevel=WORKER_LOGLEVEL,  # type: Union[str, int]
        logfile=None,  # type: str
        perform_ping_check=True,  # type: bool
        ping_task_timeout=10.0,  # type: float
        shutdown_timeout=10.0,  # type: float
        **kwargs  # type: Any
    ):
        # type: (...) -> Iterable
        """Start embedded worker.
    
        Yields:
            celery.app.worker.Worker: worker instance.
        """
        test_worker_starting.send(sender=app)
    
        try:
            with _start_worker_thread(app,
                                      concurrency=concurrency,
                                      pool=pool,
                                      loglevel=loglevel,
                                      logfile=logfile,
                                      perform_ping_check=perform_ping_check,
                                      shutdown_timeout=shutdown_timeout,
                                      **kwargs) as worker:
                if perform_ping_check:
                    from .tasks import ping
                    with allow_join_result():
                        assert ping.delay().get(timeout=ping_task_timeout) == 'pong'
    
                yield worker
        finally:
>           test_worker_stopped.send(sender=app, worker=worker)
E           UnboundLocalError: local variable 'worker' referenced before assignment
/usr/local/lib/python3.7/site-packages/celery/contrib/testing/worker.py:91: UnboundLocalError
```

--- a/celery/contrib/testing/worker.py
+++ b/celery/contrib/testing/worker.py
@@ -72,6 +72,7 @@ def start_worker(
     """
     test_worker_starting.send(sender=app)
 
+    worker = None
     try:
         with _start_worker_thread(app,
                                   concurrency=concurrency,

